extensions/2.0/Vendor/MAXAR_collision_geometry/README.md

```diff
@@ -0,0 +1,82 @@
+# MAXAR_collision_geometry
+
+## Contributors
+
+* Erik Dahlström, Maxar, [@erikdahlstrom](https://github.com/erikdahlstrom)
+* Sam Suhag, Cesium, [@sanjeetsuhag](https://github.com/sanjeetsuhag)
+
+## Status
+
+Draft
+
+## Dependencies
+
+Written against the glTF 2.0 spec.
+
+## Overview
+
+In gaming and simulation engines, high resolution 3D models are often paired with decimated versions of the model that are optimized for use in rigid-body physics. These collision volumes are the union of a set of convex polyhedra. This extension to glTF enables designating meshes as collision volumes for other meshes.
+
+## Adding Collision Meshes
+
+The following example illustrates how a mesh can designate another mesh as its collision volume:
+
+```json
+"meshes": [
+  {
+    "name": "Detailed Car Mesh",
+    "primitives": [
+      {
+        "attributes": {
+          "POSITION": 0,
+          "NORMAL": 1
+        },
+        "indices": 2
+      }
+    ],
+    "extensions": {
+      "MAXAR_collision_geometry": {
+        "mesh": 1
+      }
+    }
+  },
+  {
+    "name": "Box Collision Mesh",
+    "primitives": [
+      {
+        "attributes": {
+          "POSITION": 3,
+        },
+        "indices": 4
+      }
+    ]
+  },
+]
+```
+
+### Mesh Requirements
+
+This extension enforces the following rules on the relationships between meshes:
+
+1. Two or more meshes may use the same mesh as a collision mesh.
+2. A mesh may not designate itself as its collision mesh.
```

Comment:
I think the concern is mostly to separate the collision mesh from the visual mesh. There might be some ambiguity in what operations should be applied to the collision mesh vs the normal mesh, and the hope is that the separation will lead to fewer misunderstandings.

Reply:
I agree that this is an artificial limitation, there's no reason to disallow using itself as its own collision mesh.

In a scene there's often a mix of geometry types:

1. No collisions
2. Simplified collision mesh
3. Geometry that's already so simple that there's nothing to take away.

A textured die would be the most obvious example of (3)

As it stands, I'd have to create a duplicate cube mesh purely to put around my textured die, as otherwise I couldn't tell the difference between (1) and (3)

Reply:
I agree in principle, but duplicating a simple mesh might not be so bad?

In the future there might be additional data that only applies to the collision mesh (e.g custom data textures, vertex attributes etc), and by having clear separation this will be easier to deal with.

Reply:
The issue is that _preventing_ a mesh from being marked as being its own collision is an artificial limitation that makes this extension effectively unusable for many applications.

For example, in my application, the vast majority of meshes are their own collision mesh.
The "no-collide" and "simplified collide" cases are rare (perhaps 1 in 1000 meshes).

At present, using this extension would double the size of the glTF in almost all cases.
Furthermore, my application would need to de-duplicate the meshes when loading to spot that the visual and collision meshes are identical (otherwise there would be a large performance cost). I thus could not use this extension.

If I could instead mark the visual and collision meshes as being the same one, then this extension would fulfill all my current and expected requirements.

Permitting a mesh to be its own collision mesh clearly has no effect on an application that needs to provide different data for the collision and visual meshes, as by definition they would be different meshes.

Reply:
For another example, the [`dedup()`](https://gltf-transform.donmccurdy.com/functions.html#dedup) function in glTF Transform detects and consolidates identical resources in a glTF file. I expect that other optimization tools like gltfpack do the same, and would need special exceptions for this case.

As @RichardTea mentions, it's easier and more efficient to do this de-duplication offline than online.

If future additions require putting different data (vertex attributes, materials) onto the collision mesh, that's certainly a justifiable case for using two meshes – but I think it's better to only require duplication for inherent differences in the mesh content, not based on the external usage of the mesh.

That said – this is just a suggestion. Because this is currently a vendor extension, you don't need technical review or approval here. If you're happy with the extension and have begun using it in its current form, this PR can be merged as-is.

Reply:
I'm open to changing this, noting that with the datasets we have right now, we have no cases that would benefit size-wise from removing the restriction that I'm aware of.

We are already using the extension in its current form, and if we do change this it will need to be done in some versioned fashion.

How would you suggest relaxing the current restriction?